src/Search/SearchResult.php

```diff
@@ -37,11 +43,24 @@ class SearchResult implements \Countable, \IteratorAggregate
 
     public function __construct(array $body)
     {
+        if (isset($body['estimatedTotalHits'])) {
+            $this->numberedPagination = false;
+
+            $this->offset = $body['offset'];
+            $this->limit = $body['limit'];
+            $this->estimatedTotalHits = $body['estimatedTotalHits'];
+            $this->hitsCount = \count($body['hits']);
+        } else {
+            $this->numberedPagination = true;
+
+            $this->hitsPerPage = $body['hitsPerPage'];
+            $this->page = $body['page'];
+            $this->totalPages = $body['totalPages'];
+            $this->totalHits = $body['totalHits'];
+            $this->hitsCount = $body['totalHits'];
```

Comment:
In one case `hitsCount` is the length of the hits array, and in the other, the `totalHits`?Is this what we want?

Reply:
The `hitsCount` using the array's length was done in the past. This is to ensure backward compatibility.